infrastructure/kubernetes-demo/template.yaml

```diff
@@ -0,0 +1,244 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+
+Description: "A stack to run a demo container in EKS"
+
+Parameters:
+  ServiceName:
+    Type: String
+    Default: hello-node-eks
+  KubernetesVersion:
+    Type: String
+    Default: 1.21
+  CommonStackName:
+    Type: String
+    Default: common
+  Namespace:
+    Type: String
+    Default: hello-world
+
+Resources:
+  HelloWorldEKSCluster:
+    Type: AWS::EKS::Cluster
+    Properties:
+      Name: !Join ['', [!Ref ServiceName, Cluster]]
+      ResourcesVpcConfig:
+        EndpointPrivateAccess: true
+        SubnetIds:
+        - Fn::ImportValue:
+            Fn::Sub: "${CommonStackName}-PublicSubnet1ID"
+        - Fn::ImportValue:
+            Fn::Sub: "${CommonStackName}-PublicSubnet2ID"
+      RoleArn: !GetAtt KubernetesRole.Arn
+      Version: !Ref KubernetesVersion
+
+  HelloWorldEKSFargateProfile:
+    Type: AWS::EKS::FargateProfile
+    Properties:
+      ClusterName: !Ref HelloWorldEKSCluster
+      PodExecutionRoleArn: !GetAtt PodExecutionRole.Arn
+      Selectors:
+        - Namespace: !Ref Namespace
+      Subnets:
+        - Fn::ImportValue:
+            Fn::Sub: "${CommonStackName}-PrivateSubnet1ID"
+        - Fn::ImportValue:
+            Fn::Sub: "${CommonStackName}-PrivateSubnet2ID"
+
+  KubeSystemEKSFargateProfile:
+    Type: AWS::EKS::FargateProfile
+    Properties:
+      ClusterName: !Ref HelloWorldEKSCluster
+      PodExecutionRoleArn: !GetAtt PodExecutionRole.Arn
+      Selectors:
+        - Namespace: kube-system
+      Subnets:
+        - Fn::ImportValue:
+            Fn::Sub: "${CommonStackName}-PrivateSubnet1ID"
+        - Fn::ImportValue:
+            Fn::Sub: "${CommonStackName}-PrivateSubnet2ID"
+
+  PodExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Join ['', [!Ref ServiceName, PodExecutionRole]]
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: eks-fargate-pods.amazonaws.com
+            Action: 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonEKSFargatePodExecutionRolePolicy
+
+  KubernetesRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Join ['', [!Ref ServiceName, KubernetesRole]]
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: eks-fargate-pods.amazonaws.com
+            Action: 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonEKSClusterPolicy
+
+  AWSLoadBalancerControllerIAMPolicy:
```

Comment:
Urgh this thing... this was the JSON right?
It's so big 😭

Reply:
Yup, totally agree. I'm not sure why AWS don't provide this as a default managed policy like the similar one on line 85. 

But this does make me think about the structure of our CloudFormation and what can / should go in the common stack. I went with one common stack and then one stack per deployed 'thing' because in my head that's easier to reason about, but in Terraform you tend to have one `.tf` file per service you use which is also a useful model. 

So in our case we'd have a VPC stack, an IAM stack, an EKS stack etc. and that might help with hiding these ugly policy definitions away from the parts of the code we interact with frequently.

I'm not sure that an early prototype is the right time and place to be worrying too much about refactoring our CloudFormation (although maybe it is - we are supposed to be learning while doing it!) but we should definitely have a good think about how this structure has worked for us and how we might want to change it for a live service afterwards.

Reply:
We should put a shout out in the #di-tech channel on slack. I'd be interestd in seeing how others are doing their CF here. 
Perhaps some folks who've been deep in amazon for a while like @philandstuff or Paul J will have some thoughts on best pratices 🤔

Could make an interesting show and tell?

Reply:
CloudFormation has the concept of [Stack Sets](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stacksets-concepts.html) which seem to be a sort of 'stack of stacks' which could make the mental model easier to manage if we did decide to go with one stack per service that's managed by parameters defined in the stack set.

Although AWS also say that you should ['Organize your stacks by lifecycle and ownership'](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/best-practices.html#organizingstacks) which is roughly how we have it set up at the moment.

Reply:
[I put a shout out here](https://gds.slack.com/archives/CAC6S0UHK/p1651052397087429)